server/models/Transaction.js

```diff
@@ -411,19 +411,54 @@ export default (Sequelize, DataTypes) => {
       transaction.amountInHostCurrency = Math.round(transaction.amountInHostCurrency);
     }
 
-    const oppositeTransaction = {
-      ...transaction,
-      type: -transaction.amount > 0 ? TransactionTypes.CREDIT : TransactionTypes.DEBIT,
-      FromCollectiveId: transaction.CollectiveId,
-      CollectiveId: transaction.FromCollectiveId,
-      HostCollectiveId: await models.Collective.getHostCollectiveId(transaction.FromCollectiveId), // see https://github.com/opencollective/opencollective/issues/1154
-      amount: -transaction.netAmountInCollectiveCurrency,
-      netAmountInCollectiveCurrency: -transaction.amount,
-      amountInHostCurrency: Math.round(-transaction.netAmountInCollectiveCurrency * transaction.hostCurrencyFxRate),
-      hostFeeInHostCurrency: transaction.hostFeeInHostCurrency,
-      platformFeeInHostCurrency: transaction.platformFeeInHostCurrency,
-      paymentProcessorFeeInHostCurrency: transaction.paymentProcessorFeeInHostCurrency,
-    };
+    // Is the target "collective" (account) "Active" (has an host, manage its own budget)
+    const fromCollective = await models.Collective.findByPk(transaction.FromCollectiveId);
+
+    let oppositeTransaction;
+    if (!fromCollective.isActive) {
+      oppositeTransaction = {
+        ...transaction,
+        type: -transaction.amount > 0 ? TransactionTypes.CREDIT : TransactionTypes.DEBIT,
+        FromCollectiveId: transaction.CollectiveId,
+        CollectiveId: transaction.FromCollectiveId,
+
+        HostCollectiveId: null,
+
+        amount: -transaction.netAmountInCollectiveCurrency,
+        netAmountInCollectiveCurrency: -transaction.amount,
+
+        amountInHostCurrency: Math.round(-transaction.netAmountInCollectiveCurrency * transaction.hostCurrencyFxRate),
+        hostFeeInHostCurrency: transaction.hostFeeInHostCurrency,
+        platformFeeInHostCurrency: transaction.platformFeeInHostCurrency,
+        paymentProcessorFeeInHostCurrency: transaction.paymentProcessorFeeInHostCurrency,
+      };
+    } else {
+      const oppositeTransactionCurrencyFxRate = await getFxRate(
```

Comment:
I don't think calling `getFxRate` is that expensive, perhaps we should apply this logic even in the case the collective is not active and simplify this function even further. I'm fine with both.